test/SafeProtocolMediator.spec.ts

```diff
@@ -7,60 +7,91 @@ import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
 import { buildRootTx, buildSingleTx } from "./utils/builder";
 
 describe("SafeProtocolMediator", async () => {
-    let deployer: SignerWithAddress, user1: SignerWithAddress;
+    let deployer: SignerWithAddress, owner: SignerWithAddress, user1: SignerWithAddress, user2: SignerWithAddress;
 
     before(async () => {
-        [deployer, user1] = await hre.ethers.getSigners();
+        [deployer, owner, user1] = await hre.ethers.getSigners();
     });
 
-    async function deployContractFixture() {
-        const safeProtocolMediator = await (await hre.ethers.getContractFactory("SafeProtocolMediator")).deploy();
+    async function deployContractsFixture() {
+        [deployer, owner, user1, user2] = await hre.ethers.getSigners();
+        const safeProtocolRegistry = await hre.ethers.deployContract("SafeProtocolRegistry", [owner.address]);
         const safe = await hre.ethers.deployContract("TestExecutor");
-        return { safeProtocolMediator, safe };
+        const safeProtocolMediator = await (
+            await hre.ethers.getContractFactory("SafeProtocolMediator")
+        ).deploy(owner.address, await safeProtocolRegistry.getAddress());
+        return { safeProtocolMediator, safeProtocolRegistry, safe };
     }
 
     describe("Setup mediator", async () => {
         it("Should set mediator as a module for a safe", async () => {
             const safe = await hre.ethers.deployContract("TestExecutor");
-            const { safeProtocolMediator } = await loadFixture(deployContractFixture);
+            const { safeProtocolMediator } = await loadFixture(deployContractsFixture);
             expect(await safe.setModule(await safeProtocolMediator.getAddress()));
         });
     });
 
     describe("Modules", async () => {
-        async function deployContractsFixture() {
-            const { safeProtocolMediator, safe } = await loadFixture(deployContractFixture);
+        async function deployContractsWithModuleFixture() {
+            const { safeProtocolMediator, safe, safeProtocolRegistry } = await loadFixture(deployContractsFixture);
             const module = await (await hre.ethers.getContractFactory("TestModule")).deploy();
-            return { safeProtocolMediator, safe, module };
+            await safeProtocolRegistry.connect(owner).addComponent(module);
+            return { safeProtocolMediator, safe, module, safeProtocolRegistry };
         }
 
+        describe("Enabling module reverts with ModuleNotPermitted", async () => {
```

Comment:
Why does it need a separate describe block instead of being placed in the "test enable module"? IMO "Test enable module" should test for both happy and unhappy scenarios